static/src/javascripts/projects/common/modules/identity/email-preferences.js

```diff
@@ -0,0 +1,122 @@
+// This is a workaround for the email preferences page https://profile.thegulocal.com/email-prefs
+// We want to submit subscribe/unsubscribe requests without a full page refresh
+// Hopefully this will be short-lived; if it is still alive in 2017, git blame and cry
+
+define([
+    'bean',
+    'qwery',
+    'reqwest',
+    'fastdom',
+    'common/utils/$'
+], function (
+    bean,
+    qwery,
+    reqwest,
+    fastdom,
+    $
+) {
+    function useReqwest(buttonEl) {
+        bean.on(buttonEl, 'click', function () {
+            buttonEl.disabled = true;
+            buttonEl.innerHTML = 'Loading...';
+            var formQueryString = generateFormQueryString(buttonEl);
+            reqwest({
+                url: '/email-prefs',
+                method: 'POST',
+                data: formQueryString,
+                error: function () {
+                    renderErrorMessage(buttonEl);
+                },
+                success: function (response) {
+                    var subscriptionState;
+                    try {
```

Comment:
rather than a try-catch, a better way to delve into an object that you're not sure has the structure you need is:

`if (response && response.subscriptions && response.subscriptions.length)`

However I'm not sure we actually need `subscriptionState` at all. Maybe we should just flip the button state if we get success, and leave it as is if we get an error? Rather than relying on the server response to determine whether we were subscribing or unsubscribing